transit_odp/organisation/tests/csv/test_service_codes.py

```diff
@@ -121,92 +138,308 @@ def test_registered_service_code():
     otc_lic1 = LicenceModelFactory(number=licence_number)
     ServiceModelFactory(
         licence=otc_lic1,
-        registration_number=service_code,
+        registration_number=service_code.replace(":", "/"),
         service_number=service_number,
     )
 
     service_codes_csv = ServiceCodesCSV(org1.id)
 
     queryset = service_codes_csv.get_queryset()
-    assert queryset.count() == 1
-    row = service_codes_csv.get_row(queryset.first())
-    assert row[0] == ServiceCodesCSV.REGISTERED
-    assert row[1] == licence_number
-    assert row[2] == service_code
-    assert row[3] == service_number
-    assert row[4] in [f"{dataset1.id};{dataset2.id}", f"{dataset2.id};{dataset1.id}"]
+    assert len(queryset) == 1
+    row = queryset[0]
+    assert row["licence_number"] == licence_number
+    assert row["service_code"] == service_code.replace(":", "/")
+    assert row["line_number"] == service_number
+    # only 1 service code, 2 TXCFileAttributes
+    assert row["dataset_id"] == dataset2.id
```

Comment:
yes correct! the behaviour has changed, in the original code we had this method:

``` 
# transit_odp/organisation/csv/service_codes.py

-    def to_csv(self):
-        string_ = io.StringIO()
-        writer = csv.writer(string_, quoting=csv.QUOTE_ALL)
-        writer.writerow(
-            ("Service Status", "Licence Number", "Service Code", "Line", "Dataset ID")
-        )

-        for otc_service in self.get_queryset():
-            row = self.get_row(otc_service)
-            if row[0] == self.MISSING_DATA:
-                writer.writerow(row)
-
-        for otc_service in self.get_queryset():
-            row = self.get_row(otc_service)
-            if row[0] == self.REGISTERED:
-                writer.writerow(row)
-
-        string_.seek(0)
-        return string_
```
returned the `string_ => io.StringIO`, now to be consistent with the other csvs in BODS, I am using the `CVSBuilder` in this class there is a method `to_string()` return a string  
```
        output = csvfile.getvalue()
        csvfile.close()
        logger.info(prefix + "File successfully closed")
        return output
```
output here is a string

Reply:
Actually my question was about the dataset IDs. It used to be a comma-separated list, but now it's not. I don't see anything in the ticket which makes it ok to break that.

I wanted your opinion on how long it would take to fix this regression.

Reply:
Sorry Angus Are talking about this one?

![Screenshot 2023-03-01 at 12 21 43](https://user-images.githubusercontent.com/111745616/222139459-1921cb33-9aaa-4033-912c-64e75ef04701.png)

it's `semicolon-list`

Reply:
So the output *is* a semi-colon-separated list?

The reason I thought it wasn't was because you had a unit test that output only one of the two dataset IDs.

Reply:
Actually I am asking to you, Is this `semicolon-separated list` which are you talking about?

In my test I am testing the `only one` `dataset ID` in BODS that come from `revision`.
How many dataset IDs is suppose to test for you?